### ISSUE TITLE
States not saved

### ISSUE THREAD

**Issue by SpySauce** [HabitRPG/habitica#1013]
I've been losing experience, items, and streak bonuses because often my progress is not saved whenever I exit the page or when I refresh. This doesn't happen all the time, but fairly often. Anyone else experiencing this?

**Comment by wizonesolutions** [comment]
Yeah; I load the page, make my changes ASAP, then refresh to confirm they stuck, and then I close the page.

**Comment by SpySauce** [comment]
I do that when I worry about it, but I've lost a few items this way and reconfirming the changes doesn't mean I'll get them back.

**Comment by wizonesolutions** [comment]
Yeah... @lefnire what's the root cause on this? Would it be possible to add a subtle progress indicator somewhere that turns into a green check mark or something when the client is up to date with the server? Assuming that it's possible to check this.

**Comment by Elkian** [comment]
I've been having the same problem, often. And when I try to refresh I usually get an error that keeps me off the page for several hours. The last time it failed to boot, I logged out, and now my account doesn't exist. :\

**Comment by lefnire** [comment]
see http://habitrpg.tumblr.com/post/51082061058/whats-with-the-downtime , we're scrambling to fix this

**Comment by weelillad** [comment]
I second wizonesolutions's suggestion for a progress/sync indicator on the page. I think it really degrades the user experience when I can't ever be sure whether my changes have stuck or not. Just lost a 2-week streak due to changes not sticking, so I'm rather upset right now..

**Comment by lefnire** [comment]
you can restore streaks via Task Edit > Advanced > Restore Streak. A sync
indicator won't help us, since it's bugs in the framework and there's no
way of our telling whether or not a supposedly successful transaction was
actually successul. Our best bet now is the Racer 0.5 migration, which
we'll be doing in the next week or two.

**Comment by weelillad** [comment]
Thanks for the quick reply and the advice! I'm still thinking that some indicator might be helpful, although maybe not for this particular problem. I'll continue to think about it as I use HabitRPG, and will speak up when I have more concrete suggestions. =)

**Comment by aurakami** [comment]
I just had this happen to me. I finally got to level two (after leaving at 147/150 xp last night) at 12/160 xp. I just refreshed and it took me back down to 147/150 and it deducted my two coins (had 7, now 5) but still kept the dailies I did so far today (to get me to level two) checked. I tried unchecking and rechecking them which didn't work (148/150) and then refreshed and it took me down to 140/150 and removed another coin (4/7) while still leaving what I did today checked.
